src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -197,7 +197,7 @@ public void Enqueue(Event e)
                 break;
             } while (true);
 
-            Thread.Sleep(1);  // give EventPump thread a chance to process the event
+            Thread.Sleep(0);  // give EventPump thread a chance to process the event
```

Comment:
Before we fix this I'd like to have a good idea of why it was important. `Thread.Sleep(1)` forces a context switch and `Thread.Sleep(0)` does not force a context switch. We may be losing the ability for the even pump to process an event. The original change from `0` to `1` should have come with a test demonstrating the necessity, but I think we should at least try to figure it out and add that test now if possible. There may be a better way to solve the original problem than either `Thread.Sleep(1)` or `Thread.Sleep(0)`.